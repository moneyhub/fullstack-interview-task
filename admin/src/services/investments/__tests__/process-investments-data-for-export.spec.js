
// necessary because jest config is setup to reset mocks after every test
function setupMocks() {
  jest.mock("../../financial-data/get-holding-name-memoized", () =>{
    return {
      getHoldingNameMemoized: jest.fn().mockReturnValue("generic holding"),
    }
  })
}

describe("processInvestmentsDataForExport", () => {
  beforeEach(() => {
    setupMocks()
  })

  it("generates the expected data for a given input", async () => {
    const {mockInvestments} = require("../../../__tests__/testData")
    const {processInvestmentsDataForExport} = require("../process-investments-data-for-export")

    const result = await processInvestmentsDataForExport(mockInvestments)

    expect(JSON.stringify(result, null, 2)).toMatchSnapshot()
  })

  it("calls getHoldingNameMemoized with the correct parameter for each holding for each record", async () => {
    const {mockInvestments} = require("../../../__tests__/testData")
    const {processInvestmentsDataForExport} = require("../process-investments-data-for-export")

    const input = [
      mockInvestments[0],
      mockInvestments[1],
    ]

    await processInvestmentsDataForExport(input)

    const {getHoldingNameMemoized} = require("../../financial-data/get-holding-name-memoized")

    expect(getHoldingNameMemoized).toHaveBeenNthCalledWith(1, "2")
    expect(getHoldingNameMemoized).toHaveBeenNthCalledWith(2, "1")
    expect(getHoldingNameMemoized).toHaveBeenNthCalledWith(3, "2")
  })
})
