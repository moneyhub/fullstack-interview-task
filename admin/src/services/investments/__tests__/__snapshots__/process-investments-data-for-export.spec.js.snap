// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processInvestmentsDataForExport generates the expected data for a given input 1`] = `
"[
  {
    \\"First Name\\": \\"Billy\\",
    \\"Last Name\\": \\"Bob\\",
    \\"Date\\": \\"2020-01-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 1400
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-01-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 10000
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-01-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 10000
  },
  {
    \\"First Name\\": \\"Billy\\",
    \\"Last Name\\": \\"Bob\\",
    \\"Date\\": \\"2020-02-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 1300
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-02-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 11000
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-02-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 11000
  },
  {
    \\"First Name\\": \\"Billy\\",
    \\"Last Name\\": \\"Bob\\",
    \\"Date\\": \\"2020-03-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 12000
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-03-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 10750
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-03-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 6450
  },
  {
    \\"First Name\\": \\"Sheila\\",
    \\"Last Name\\": \\"Aussie\\",
    \\"Date\\": \\"2020-03-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 4300
  },
  {
    \\"First Name\\": \\"John\\",
    \\"Last Name\\": \\"Smith\\",
    \\"Date\\": \\"2020-03-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 120000
  },
  {
    \\"First Name\\": \\"John\\",
    \\"Last Name\\": \\"Smith\\",
    \\"Date\\": \\"2020-03-01\\",
    \\"Holding\\": \\"generic holding\\",
    \\"Value\\": 30000
  }
]"
`;
